operator/pkg/custom/client.go

```diff
@@ -2,27 +2,43 @@ package custom
 
 import (
 	"context"
+	"fmt"
+	"github.com/kyma-project/manifest-operator/operator/pkg/labels"
 	"github.com/kyma-project/manifest-operator/operator/pkg/util"
 	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	k8slabels "k8s.io/apimachinery/pkg/labels"
 	"k8s.io/client-go/rest"
 	"sigs.k8s.io/controller-runtime/pkg/client"
-	"sigs.k8s.io/controller-runtime/pkg/cluster"
 )
 
 type ClusterClient struct {
 	DefaultClient client.Client
 }
 
-func (cc *ClusterClient) GetNewClient(restConfig *rest.Config) (client.Client, error) {
-	cluster, err := cluster.New(restConfig)
+func (cc *ClusterClient) GetNewClient(restConfig *rest.Config, options client.Options) (client.Client, error) {
+	client, err := client.New(restConfig, options)
 	if err != nil {
 		return nil, err
 	}
-	return cluster.GetClient(), nil
+	return client, nil
 }
 
 func (cc *ClusterClient) GetRestConfig(ctx context.Context, kymaOwner string, namespace string) (*rest.Config, error) {
-	kubeConfigSecret := v1.Secret{}
+	kubeConfigSecretList := &v1.SecretList{}
+	if err := cc.DefaultClient.List(ctx, kubeConfigSecretList, &client.ListOptions{
+		LabelSelector: k8slabels.SelectorFromSet(
+			k8slabels.Set{labels.ComponentOwner: kymaOwner}), Namespace: namespace,
+	}); err != nil {
+		return nil, err
+	} else if len(kubeConfigSecretList.Items) < 1 {
+		gr := v1.SchemeGroupVersion.WithResource("secrets").GroupResource()
```

Comment:
I think we need a better way for this stuff :D; considering youre interacting with the API Server its safe to use the GroupVersion method of the kubeConfigSecretList variable as it was filled by the API Client.

Reply:
tried to improve that